src/window/floatingwindow.cpp

```diff
@@ -66,3 +69,52 @@ void FloatingWindowInterface::onReload(QObject* oldInstance) {
 }
 
 ProxyWindowBase* FloatingWindowInterface::proxyWindow() const { return this->window; }
+
+void FloatingWindowInterface::onVisibilityChanged() {
+	auto* qw = this->window->backingWindow();
+	auto visibility = qw ? qw->visibility() : QWindow::Hidden;
+
+	auto maximized = visibility == QWindow::Maximized;
+	auto fullscreen = visibility == QWindow::FullScreen;
+
+	if (maximized != this->mMaximized) {
+		this->mMaximized = maximized;
+		emit this->maximizedChanged();
+	}
+
+	if (fullscreen != this->mFullscreen) {
+		this->mFullscreen = fullscreen;
+		emit this->fullscreenChanged();
+	}
+}
+
+bool FloatingWindowInterface::maximized() const { return this->mMaximized; }
+bool FloatingWindowInterface::fullscreen() const { return this->mFullscreen; }
```

Comment:
We can store the maximized or fullscreen flags for application before its visible, but should reflect the actual window state when queried (as well as be notified when it changes)

Reply:
Actually connected to windowStateChanged, not visibility*